In [2]:
from scipy.stats import chisquare

observed = [21307, 21306, 21387]
expected = [64000 / 3] * 3

result = chisquare(f_obs=observed, f_exp=expected)

print(f"Chi-square statistic: {result.statistic:.6f}")
print(f"P-value: {result.pvalue:.6f}")

Chi-square statistic: 0.202531
P-value: 0.903693


### SRM Test Results

- Chi-square statistic: 0.202531
- Degrees of freedom: 2
- P-value: 0.903693
- Significance level: 0.05

**Interpretation:** We fail to reject the null hypothesis.
There is no statistically significant evidence of a sample
ratio mismatch under the assumed 1:1:1 allocation.


### Covariate Balance Assessment

Pre-treatment customer characteristics were compared across
the three randomized groups using standardized mean
differences (SMDs).

| Covariate | Men's Email vs Control | Women's Email vs Control |
|---|---:|---:|
| History | 0.0076 | 0.0065 |
| Recency | 0.0068 | 0.0052 |
| Newbie | -0.0009 | 0.0026 |
| Men's merchandise history | -0.0046 | -0.0086 |
| Women's merchandise history | 0.0076 | 0.0049 |

**Interpretation:** All evaluated covariates have absolute
SMDs below 0.01, indicating negligible observed
standardized differences between treatment and control groups.

Together with the SRM test, these findings provide no
obvious evidence of allocation imbalance in the evaluated
variables. They do not establish that randomization was
perfect or that all potential imbalances have been ruled out.



### Categorical Covariate Balance

The distributions of purchase channel (`channel`) and
geographic category (`zip_code`) were compared across
the three experimental groups.

**Purchase Channel Distribution**

| Channel | Men's Email | Control | Women's Email |
|---|---:|---:|---:|
| Multichannel | 12.09% | 12.23% | 12.06% |
| Phone | 43.37% | 43.78% | 44.20% |
| Web | 44.54% | 43.99% | 43.74% |

**Geographic Distribution**

| Category | Men's Email | Control | Women's Email |
|---|---:|---:|---:|
| Rural | 15.22% | 14.73% | 14.87% |
| Surburban | 44.59% | 45.18% | 45.12% |
| Urban | 40.19% | 40.09% | 40.01% |

**Interpretation**

- The largest channel difference relative to control was
  0.55 percentage points (Men's Email, Web).
- The largest geographic difference relative to control was
  0.59 percentage points (Men's Email, Surburban).
- No substantial descriptive imbalance was observed
  in either categorical variable.

**Limitation:** These comparisons are descriptive rather
than formal statistical tests. Observed balance does not
guarantee perfect randomization or rule out unobserved
imbalances.
